stbt_auto_selftest.py

```diff
@@ -92,23 +96,33 @@ def main(argv):
     os.chdir(root)
 
     if cmdline_args.command == 'generate':
-        generate()
+        generate(cmdline_args.source_files)
     elif cmdline_args.command == 'validate':
         return validate()
     else:
         assert False
 
 
-def generate():
-    tmpdir = generate_into_tmpdir()
+def generate(source_files):
+    valid_source_files = _valid_source_files(source_files)
+    tmpdir = generate_into_tmpdir(valid_source_files)
     try:
-        target = "%s/selftest/auto_selftest" % os.curdir
-        if os.path.exists(target):
-            shutil.rmtree(target)
-        os.rename(tmpdir, target)
-    except:
-        shutil.rmtree(tmpdir)
-        raise
+        if source_files:
+            for f in valid_source_files:
+                target = os.path.join(os.curdir, "selftest/auto_selftest",
+                                      _selftest_filename(f))
+                if os.path.exists(target):
+                    os.remove(target)
```

Comment:
I think I just got confused -- at some point I thought os.rename would fail if target file existed, but it must've been a permissions issue or something else. I've removed this.